datadriven/src/line_sparser.rs

```diff
@@ -0,0 +1,115 @@
+use crate::test_data::CmdArg;
+use anyhow::Result;
+use regex::Regex;
+
+// token
+// (1) argument
+// (2) argument=a,b,c
+// (3) argument=
+// (4) argument=(a,b,c,...)
+// (5) a,b,c
+pub fn parse_line(line: &str, logger: &slog::Logger) -> Result<(String, Vec<CmdArg>)> {
+    let field = split_directives(line)?;
+    if field.is_empty() {
+        bail!("empty lines occurs, unexpected.");
+    }
+
+    debug!(logger, "argument after split: {:?}", field);
+
+    let cmd = field[0].clone();
+    let mut cmd_args = vec![];
+
+    for arg in field[1..].iter() {
+        let vs: Vec<String> = arg.split_terminator('=').map(|v| v.to_string()).collect();
+        if vs.len() == 1 {
+            cmd_args.push(CmdArg {
+                key: vs[0].clone(),
+                values: vec![],
+            })
+        } else if vs.len() == 2 {
+            let (key, value) = (vs[0].clone(), vs[1].clone());
+            let mut values: Vec<String>;
+            if value.starts_with('(') && value.ends_with(')') {
+                values = value[1..value.len() - 1]
+                    .split_terminator(',')
+                    .map(|v| v.to_string())
+                    .collect();
+            } else {
+                values = value.split_terminator(',').map(|v| v.to_string()).collect();
+            }
+            values = values.into_iter().map(|v| v.trim().to_string()).collect();
+            cmd_args.push(CmdArg { key, values })
+        } else {
+            bail!("unknown argument format: {}", arg)
+        }
+    }
+
+    Ok((cmd, cmd_args))
+}
+
+lazy_static! {
+    static ref RE: Regex =
+        Regex::new(r"^ *[-a-zA-Z0-9/_,.]+(=[-a-zA-Z0-9_@=+/,.]*|=\([^)]*\)| *)( |$)").unwrap();
```

Comment:
I don't think `/` is allowed.

Reply:
They have
 ```var splitDirectivesRE = regexp.MustCompile(`^ *[-a-zA-Z0-9/_,\.]+(|=[-a-zA-Z0-9_@=+/,\.]*|=\([^)]*\))( |$)`)```,
 however I think `a-z` `A-Z` `_` are enough.

Reply:
Turned out I had compared with an old version of the file, which is used by etcd. `/` is allowed actually.

But why allow multiple space as indicated by `| *`?

Reply:
That is a workaround to catch type `key`, because `rust` can't catch `empty`, so I use white space instead

Reply:
I think it's because you remove the first `|`. If you keep the original go regex expression, it should match pure key correctly.